### ISSUE TITLE
Multiple selection doesn't work in InteliJ-based IDEs

### ISSUE THREAD

**Issue by Ryvix** [carlrobertoh/CodeGPT#123]
Hello,

If I hold alt and click I can select multiple things but when prompting GPT it only sends the last selected one to the chat.

It should instead allow to send all selected items to the chat.

However when replacing the resulting code or the output of it, it would have to use a character as a separator. I think it's important to not just use a line break or a space since there might not be that character in the original selection itself. Therefore, maybe there could be options for the separator character.

It could have options to use a line break or space or other character. Perhaps if there was a radio button to choose to use a line break and one to choose a space for the separator character. And a box to enter a different character somehow. Not sure if it should optionally have the ability to enter a character code or something of that nature. This would be better than just connecting them with nothing and probably a line break because they wouldn't necessarily always have a line break in the selections.

Also maybe there could be an option to control the order it sends the selections to the chat in, either in the order they appear in the document or the order in which they are selected.

I don't know if there's some way to add an option on the popup when you edit the prompts. Maybe they would be good to be able to be easily changed there somehow or maybe there is some other way to quickly edit them I don't know about.

I really like your software, thank you!

**Comment by carlrobertoh** [comment]
Hey, sorry for the delayed response.

I'm not sure if it's possible to detect if multiple selections are highlighted in the main editor. However, if it is possible, we could display a dialog box where user can modify the payload, change the order, add prompts to each selection block, use custom delimiters, etc., before sending it to the API.

**Comment by Ryvix** [comment]
No worries.

That might be a good idea too.

What if I wanted to have a single prompt for all of them together too?

Or I was also thinking what if it could do what you're suggesting and maybe it could just send back each section to the original section when you have that replace selection button or whatever it is in the latest version.

Honestly I liked the button and I think in 2.0 I had to right click to see copy or replace options so I had just reverted back to the old version since I liked the way it worked better with opening new windows with each prompt and the buttons. But overall I really like this plugin. Thank you!
